libsupport/src/Http.cpp

```diff
@@ -127,15 +114,8 @@ HttpUploadCommon(CurlHandle&& holder, const std::string& data) {
 
 katana::Result<void>
 katana::HttpGet(const std::string& url, std::vector<char>* response) {
-  auto curl_res = CurlHandle::Make(url, response);
-  if (!curl_res) {
-    return curl_res.error();
-  }
-  CurlHandle curl(std::move(curl_res.value()));
-
-  if (auto res = curl.SetOpt(CURLOPT_HTTPGET, 1L); !res) {
-    return res.error();
-  }
+  CurlHandle curl = KATANA_CHECK(CurlHandle::Make(url, response));
```

Comment:
Just out of curiosity does `auto` work with this macro/`({...})`?

Reply:
Yes it does! For the purpose of our style guidelines though, I believe we should avoid `auto` here for the sake of clarity.

Reply:
Other than for cases where the return value is included as an explicit non-inferable argument to the call, I would recommend.